src/test/java/jenkins/scm/impl/subversion/SubversionStepTest.java

```diff
@@ -116,4 +116,52 @@ public void multipleSCMs() throws Exception {
         assertFalse(iterator.hasNext());
     }
 
+
+    @Test
+    public void checkoutDepthAsItIsInfiniteTest() throws Exception {
+        checkoutDepthTest("as-it-is-infinity");
```

Comment:
These tests check that the parameter works, but they don't test how the behaviors differ. Is it easy/possible to use `sampleRepo` to construct different repo histories so you can verify the different behavior of each parameter?

Reply:
After quick discussion it seems this is not so easy, so I am ok with just the basic tests.